app/controllers/users/backup_code_setup_controller.rb

```diff
@@ -92,7 +92,9 @@ def ensure_backup_codes_in_session
     end
 
     def generate_codes
-      revoke_remember_device(current_user)
+      if current_user.backup_code_configurations.length.nonzero?
+        revoke_remember_device(current_user)
+      end
```

Comment:
Minor: I might suggest `.any?` as a shorter alternative to `.length.nonzero?`, which would be short enough to collapse to a single line.

```suggestion
      revoke_remember_device(current_user) if current_user.backup_code_configurations.any?
```

Reply:
I like it!